gstd/gstd.c

```diff
@@ -216,12 +210,25 @@ main (gint argc, gchar * argv[])
   }
   g_option_context_free (context);
 
-  gstd_log_init (gstdlogfile, gstlogfile);
-  gstd_daemon_init (argc, argv, pidfile);
+  if (!nodaemon) {
+    if (!gstd_log_init (gstdlogfile, gstlogfile)) {
+      ret = EXIT_FAILURE;
+      goto out;
+    }
+
+    if (!gstd_daemon_init (argc, argv, pidfile)) {
+      ret = EXIT_FAILURE;
+      goto out;
+    }
+  }
+
+  gstd_debug_init();
 
   /* Print the version and exit */
   if (version) {
-    print_header(quiet);
+    if (!quiet) {
+      print_header ();
```

Comment:
Sounds good. Something that bugs me is that the version case is going through the daemon initialization when its called with "gstd -v". Couldn' t we move the print header right after the options parse and right below the version option check so that the daemon doesn' t init?